sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -285,6 +280,27 @@ TrackerT invokeNewTracker(RestrictionT restriction) {
 
     final TypeDescription clazzDescription = new TypeDescription.ForLoadedType(fnClass);
 
+    final Implementation splitRestrictionDelegation;
```

Comment:
Rather than the final variable + conditionals to set, why not:

```
final Implementation splitRestrictionDelegation = splitRestrictionDelegation(signature.splitRestriction())
```

And moving that to a helper method? If you want the possibilities here, I'd also see this as working:

```
splitRestrictionDelegation = Objects.firstNonNull(
    splitRestrictionFor(signature.splitRestriction()),
    MethodDelegaiton.to(DefaultSplitRestriction.class));
```

Where `splitRestrictionFor` returns null if it can't create the delegation.